url64d634b8f0_agdturner_generic_tgz-pJ8-uk_ac_leeds_ccg_andyt_generic_visualisation_charts_Generic_BarChartJ8/src/uk/ac/leeds/ccg/andyt/generic/visualisation/charts/Generic_Plot.java

```diff
@@ -0,0 +1,248 @@
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package uk.ac.leeds.ccg.andyt.generic.visualisation.charts;
+
+import java.awt.Rectangle;
+import java.awt.print.PrinterJob;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.util.concurrent.Future;
+import javax.print.StreamPrintService;
+import javax.print.StreamPrintServiceFactory;
+import javax.print.attribute.HashPrintRequestAttributeSet;
+import javax.print.attribute.PrintRequestAttributeSet;
+import javax.print.attribute.standard.Copies;
+import uk.ac.leeds.ccg.andyt.generic.utilities.Generic_Execution;
+import uk.ac.leeds.ccg.andyt.generic.visualisation.Generic_Canvas;
+import uk.ac.leeds.ccg.andyt.generic.visualisation.Generic_Printable;
+import uk.ac.leeds.ccg.andyt.generic.visualisation.Generic_Visualisation;
+
+/**
+ * This class acts holds a reference to a lightweight component for rendering.
+ * In part it acts as an interface and is extended by other classes for
+ * generating specific data plots.
+ */
+public class Generic_Plot extends Abstract_Generic_Plot {
+
+    public Generic_Canvas _Generic_Canvas;
+    public Future future;
+
+    public Generic_Plot() {
+    }
+
+    /**
+     *
+     */
+    @Override
+    public void run() {
+        try {
+            File theFile = getFile();
+            String theFormat = getFormat();
+            _Generic_Canvas = new Generic_Canvas();
+            _Generic_Canvas._Generic_Plot = this;
+            _Generic_Canvas._Rectangle = new Rectangle(0, 0, getDataWidth(), getDataHeight());
+
+//        Graphics2D g2 = (Graphics2D) _Generic_Canvas.getGraphics();
+//            setG2(g2);
+//            //_Generic_Canvas.paint(g2);
+//            Dimension dim = draw();
+//            _Generic_Canvas.setDimension(dim);
+            PrinterJob printerJob = PrinterJob.getPrinterJob();
+            Generic_Printable theGeneric_Printable = new Generic_Printable(_Generic_Canvas);
+            printerJob.setPrintable(theGeneric_Printable);
+
+            String psMimeType = "application/postscript";
+
+            FileOutputStream fileOutputStream = null;
+//        PrintService[] printServices = PrinterJob.lookupPrintServices();
+            StreamPrintService streamPrintService = null;
+            StreamPrintServiceFactory[] streamPrintServiceFactories
+                    = PrinterJob.lookupStreamPrintServices(psMimeType);
+            File psFile = new File(theFile.getParentFile(), theFile.getName() + ".ps");
+            System.out.println("psFile " + psFile.toString());
+            if (streamPrintServiceFactories.length > 0) {
+                try {
+                    psFile.createNewFile();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

`File.createNewFile` returns `false` if the named file already exists and an empty file is not created. Check the value returned by `createNewFile()`. Handle when the method returns `false`. Otherwise, you might silently overwrite an existing file which could lead to data loss. See the following diffs on GitHub that make this code change: [diff1](https://github.com/lingochamp/FileDownloader/commit/65069c18a4b1b4cfa758173ef49d69c6897fc77a#diff-2b6ad96df9bf9828c22f961e9e9a5b933ab2df7b2fb0f12d62c044309e1e2842L331), [diff2](https://github.com/m-szalik/javamail/commit/62b1cb45e32a998efbe6cfa09b06752869f590ae#diff-440ef5498fc2c802274cf136e7bd7d682bbc10c5976afb5af718984e24deced4L45)

Learn more: [File.createNewFile in the Java Platform, Standard Edition 8 API Specification](https://docs.oracle.com/javase/8/docs/api/java/io/File.html#createNewFile--)